osbenchmark/workload_generator/corpus.py

```diff
@@ -64,49 +69,162 @@ def extract(client, output_path, index, number_of_docs_requested=None):
 
     number_of_docs = client.count(index=index)["count"]
 
-    total_docs = number_of_docs if not number_of_docs_requested else min(number_of_docs, number_of_docs_requested)
+    total_docs = (
+        number_of_docs
+        if not number_of_docs_requested
+        else min(number_of_docs, number_of_docs_requested)
+    )
 
     if total_docs > 0:
-        logger.info("[%d] total docs in index [%s]. Extracting [%s] docs.", number_of_docs, index, total_docs)
+        logger.info(
+            "[%d] total docs in index [%s]. Extracting [%s] docs.",
+            number_of_docs,
+            index,
+            total_docs,
+        )
         docs_path = get_doc_outpath(output_path, index)
-        dump_documents(client, index, get_doc_outpath(output_path, index, "-1k"), min(total_docs, 1000), " for test mode")
+        dump_documents(
+            client,
+            index,
+            get_doc_outpath(output_path, index, "-1k"),
+            min(total_docs, 1000),
+            " for test mode",
+        )
         dump_documents(client, index, docs_path, total_docs)
         return template_vars(index, docs_path, total_docs)
     else:
-        logger.info("Skipping corpus extraction fo index [%s] as it contains no documents.", index)
+        logger.info(
+            "Skipping corpus extraction fo index [%s] as it contains no documents.",
+            index,
+        )
         return None
 
 
-def dump_documents(client, index, out_path, number_of_docs, progress_message_suffix=""):
-    # pylint: disable=import-outside-toplevel
-    from opensearchpy import helpers
+def dump_documents_range(
+    client, index, out_path, start_doc, end_doc, total_docs, progress_message_suffix=""
+):
+    """
+    Extract documents in the range of start_doc and end_doc and write to induvidual files
+
+    :param client: OpenSearch client used to extract data
+    :param index: Name of index to dump
```

Comment:
To help newcomers understand better, we can redefine this as:
`name of OpenSearch index to extract documents from`